DatadogRUM/Sources/RUMVitals/RenderLoop/RenderLoopObserver.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import UIKit
+
+internal protocol RenderLoopReader: AnyObject {
+    func stop()
+    func didUpdateFrame(link: FrameInfoProvider)
+}
+
+internal protocol RenderLoopObserver {
+    /// `RenderLoopObserver` registers the render loop reader to get updates of frame rendering.
+    /// - Parameters:
+    ///   - renderLoopReader: The render loop reader to be notified.
+    func register(_ renderLoopReader: RenderLoopReader)
+
+    /// `RenderLoopObserver` stops notifying the render loop reader.
+    /// - Parameters:
+    ///   - renderLoopReader: The render loop reader to stop receiving frame rendering updates.
+    func unregister(_ renderLoopReader: RenderLoopReader)
+}
+
+/// A class reading information from the display vsync.
+internal class DisplayLinker {
+    private var renderLoopReaders: [RenderLoopReader] = []
+    private var displayLink: FrameInfoProvider?
+    private let notificationCenter: NotificationCenter
+
+    init(notificationCenter: NotificationCenter) {
+        self.notificationCenter = notificationCenter
+
+        notificationCenter.addObserver(
+            self,
+            selector: #selector(appWillResignActive),
+            name: UIApplication.willResignActiveNotification,
```

Comment:
convention/ We should use [`ApplicationNotifications`](https://github.com/DataDog/dd-sdk-ios/blob/5cc5fcd31c1271d3aef92cd8395152ac87dcc836/DatadogInternal/Sources/Context/ApplicationNotifications.swift#L16):
```suggestion
            name: ApplicationNotifications.willResignActive,
```